addons/account/models/account_tax.py

```diff
@@ -1461,6 +1461,30 @@ def _add_tax_details_in_base_lines(self, base_lines, company):
         for base_line in base_lines:
             self._add_tax_details_in_base_line(base_line, company)
 
+    @api.model
+    def _distribute_delta_amount_smoothly(self, precision_digits, delta_amount, target_factors):
+        precision_rounding = float(f"1e-{precision_digits}")
+        amounts_to_distribute = [0.0] * len(target_factors)
+        if float_is_zero(delta_amount, precision_digits=precision_digits):
+            return amounts_to_distribute
+
+        sign = -1 if delta_amount < 0.0 else 1
+        nb_of_errors = round(abs(delta_amount / precision_rounding))
+        remaining_errors = nb_of_errors
+        for i, target_factor in enumerate(target_factors):
+            factor = target_factor['factor']
+            if not remaining_errors:
+                break
```

Comment:
@smetl I took this function as-is from your branch, but I would suggest two small changes to it:
- instead of passing a list of dicts, it seems simpler to pass a list of the factors themselves
- we could divide each factor by the sum of the factors (so that they sum to 1), that would avoid needing to normalize the factors in advance and avoid mistakes where the calling code forgets to do so.